plugins/DB.cpp

```diff
@@ -46,12 +47,29 @@ bool BedrockPlugin_DB::peekCommand(SQLite& db, BedrockCommand& command) {
         //         read-write command is mis-classified as read-only an executed in
         //         the peek, but even then we'll detect it after the fact and shut
         //         the node down.
-        const string& query = request["query"] + ";";
+        const string& query = request["query"];
         const string& upperQuery = SToUpper(STrim(query));
+        const string& nowhere = request["nowhere"];
+        const bool shouldRequireWhere = !nowhere.size() || SToUpper(nowhere) == "false";
+
+        if (!SEndsWith(upperQuery, ";")) {
+                SALERT("Query failed, query must end in ';'");
+
+                throw "502 Query failed";
+        }
+
         if (SStartsWith(upperQuery, "SELECT ")) {
             // Seems to be read-only
             SINFO("Query appears to be read-only, peeking.");
         } else {
+            if (shouldRequireWhere &&
+               (SStartsWith(upperQuery, "UPDATE") || SStartsWith(upperQuery, "DELETE")) &&
+               !SContains(upperQuery, "WHERE")) {
```

Comment:
How about ` WHERE ` with spaces at either side? So we'd match

`SELECT x FROM y WHERE z;`

But not `SELECT whereabouts FROM suspects;`